javalib/src/main/scala/java/time/Month.scala

```diff
@@ -0,0 +1,105 @@
+package java.time
+
+import java.time.temporal._
+
+final class Month private (name: String, value: Int, defaultLength: Int)
+    extends Enum[Month](name, value - 1) with TemporalAccessor
+    with TemporalAdjuster {
+
+  def getValue(): Int = value
+
+  // Not implemented
+  // def getDisplayName(style: TextStyle, locale: Locale): Locale
+
+  def isSupported(field: TemporalField): Boolean = field match {
+    case _: ChronoField => field == ChronoField.MONTH_OF_YEAR
+    case null           => false
+    case _              => field.isSupportedBy(this)
+  }
+
+  // Implemented by TemporalAccessor
+  // def range(field: TemporalField): ValueRange
+
+  // Implemented by TemporalAccessor
+  // def get(field: TemporalField): Int
+
+  def getLong(field: TemporalField): Long = field match {
+    case ChronoField.MONTH_OF_YEAR => ordinal + 1
+
+    case _: ChronoField =>
+      throw new UnsupportedTemporalTypeException(s"Field not supported: $field")
+
+    case _ => field.getFrom(this)
+  }
+
+  def plus(months: Long): Month = {
+    val offset = if (months < 0) months % 12 + 12 else months % 12
+    Month.of((ordinal + offset.toInt) % 12 + 1)
+  }
+
+  def minus(months: Long): Month = {
+    val offset = if (months < 0) months % 12 else months % 12 - 12
+    Month.of((ordinal - offset.toInt) % 12 + 1)
+  }
+
+  def length(leapYear: Boolean): Int = if (leapYear) maxLength else minLength
+
+  def minLength(): Int = defaultLength
+
+  def maxLength(): Int =
+    if (value == 2) defaultLength + 1 else defaultLength
+
+  def firstDayOfYear(leapYear: Boolean): Int =
+    if (leapYear) ((1 until value) foldLeft 1)(_ + Month.of(_).maxLength)
+    else ((1 until value) foldLeft 1)(_ + Month.of(_).minLength)
```

Comment:
There is too much duplicated code. Why not do `((1 until value) foldLeft 1)(_ + Month.of(_).defaultLength)` and add `1` if `leapYear && value >= 2`.